src/envs/stick_point.py

```diff
@@ -153,6 +164,17 @@ def simulate(self, state: State, action: Action) -> State:
             # Check if the stick is now held for the first time.
             if state.get(self._stick, "held") <= 0.5 and \
                 stick_rect.intersects(robot_circ):
+                # Check for a collision with the stick holder. The reason that
+                # we only check for a collision here, as opposed to every time
+                # step, is that we imagine the robot moving down in the z
+                # direction to pick up the stick, at which point it may
+                # collide with the stick holder. Otherwise, the robot would
+                # be high enough above the stick holder to avoid collisions.
+                holder_rect = self._object_to_geom(self._holder, state)
+                if robot_circ.intersects(holder_rect):
+                    # Immediately fail in case of collision.
+                    raise utils.EnvironmentFailure("Collided with holder.")
```

Comment:
we're still ignoring these env failures at planning time right? i think we should include the holder as an offending object here just for correctness's sake. it won't matter anyway